odoo/addons/base/tests/test_res_partner.py

```diff
@@ -138,3 +138,18 @@ def test_lang_computation_form_view(self):
         self.assertEqual(partner.lang, 'fr_FR')
         self.assertEqual(partner.child_ids.filtered(lambda p: p.name == "First Child").lang, 'de_DE')
         self.assertEqual(partner.child_ids.filtered(lambda p: p.name == "Second Child").lang, 'fr_FR')
+
+    def test_display_address_missing_key(self):
+        """ Check _display_address when some keys are missing. As a defaultdict is used, missing keys should be
+        filled with empty strings. """
+        country = self.env["res.country"].create({"name": "TestCountry", "address_format": "%(city)s %(zip)s"})
+        partner = self.env["res.partner"].create({
+            "name": "TestPartner",
+            "country_id": country.id,
+            "city": "TestCity",
+            "zip": "12345",
+        })
+        before = partner._display_address()
+        # Manually update the country address_format because placeholders are checked by create
+        self.env.cr.execute(f"UPDATE res_country SET address_format ='%(city)s %(zip)s %(nothing)s' WHERE id={country.id}")
```

Comment:
If you do the following change, your test will still pass.
```suggestion
        self.env.cr.execute(f"UPDATE res_country SET address_format ='%(zip)s %(city)s %(nothing)s' WHERE id={country.id}")
```

You should clear the cache.
And by the way, use parameters in the query...
```suggestion
        self.env.cr.execute("UPDATE res_country SET address_format ='%%(city)s %%(zip)s %%(nothing)s' WHERE id=%s", [country.id])
        self.env["res.country"].clear_caches()
```

Reply:
Why is the `clear_caches` method needed here ?

Reply:
> Why is the clear_caches method needed here ?

Because you are modifying records directly in SQL, it's needed for the ORM to fetch those changes from the DB. https://www.odoo.com/documentation/14.0/developer/reference/addons/orm.html#sql-execution

Reply:
And as pointed out by the documentation, you should actually use the `invalidate_cache` method.